pandora-server-shard/src/spaces/space.ts

```diff
@@ -554,3 +556,23 @@ export abstract class Space extends ServerRoom<IShardClient> {
 function IsTargeted(message: IClientMessage): message is { type: 'chat' | 'ooc'; parts: IChatSegment[]; to: CharacterId; } {
 	return (message.type === 'chat' || message.type === 'ooc') && message.to !== undefined;
 }
+
+function ClearOccupiedSlotsByCharacter(globalState: AssetFrameworkGlobalState, characterId: CharacterId): AssetFrameworkGlobalState {
```

Comment:
I feel this code should not be shard specific, instead moved to `withCharacter` function

Reply:
I agree. I moved this logic to `withCharacter` function and added two unit tests.